src/main_thread/tracks_store/tracks_store.ts

```diff
@@ -492,6 +528,119 @@ export default class TracksStore extends EventEmitter<ITracksStoreEvents> {
     }
   }
 
+  /**
+   * Handle the noPlayableRepresentation event, trigger an error if no fallback is possible.
+   * and can trigger event "noPlayableTrack"
+   * @param period - The period that has no playable representation
+   * @param bufferType - The media type that is not playable
+   */
+  private onNoPlayableRepresentation(period: IPeriodMetadata, bufferType: IBufferType) {
+    const periodHasAdaptationForType =
+      period.adaptations[bufferType] !== undefined &&
+      period.adaptations[bufferType].length > 0;
+    const firstPlayableAdaptation = findFirstPlayableAdaptation(period, bufferType);
+
+    if (!periodHasAdaptationForType) {
+      log.debug(
+        `TS: The period does not have adaptation for ${bufferType} there is no track to choose`,
+      );
+      return;
+    }
+
+    if (bufferType === "text") {
+      const textTypeInfo = getPeriodItem(this._storedPeriodInfo, period.id)?.[bufferType];
+      if (textTypeInfo !== undefined) {
+        const storedSettings = null;
+        textTypeInfo.storedSettings = storedSettings;
+        textTypeInfo.dispatcher?.updateTrack(storedSettings);
+      }
+      return;
+    }
+
+    if (
+      firstPlayableAdaptation === undefined &&
+      this.onTracksNotPlayableForType[bufferType] === "continue"
+    ) {
+      // audio or video is not playable, but let's continue the playback without audio
+      // or without video because of the option was set to "continue".
+      log.warn(`TS: No playable ${bufferType}, continuing without ${bufferType}`);
+      this.trigger("noPlayableTrack", {
+        trackType: bufferType,
+        period: { id: period.id, start: period.start, end: period.end },
+      });
+      // The previous event trigger could have had side-effects, so we
+      // re-check if we're still mostly in the same state
+      if (this._isDisposed) {
+        return; // Someone disposed the `TracksStore` on the previous side-effect
+      }
+    } else if (firstPlayableAdaptation === undefined) {
+      const noRepErr = new MediaError(
+        "NO_PLAYABLE_REPRESENTATION",
+        `No ${bufferType} Representation can be played`,
+        { tracks: undefined },
+      );
+      this.trigger("noPlayableTrack", {
+        trackType: bufferType,
+        period: {
+          id: period.id,
+          start: period.start,
+          end: period.end,
+        },
+      });
+      // The previous event trigger could have had side-effects, so we
+      // re-check if we're still mostly in the same state
+      if (this._isDisposed) {
+        return; // Someone disposed the `TracksStore` on the previous side-effect
+      }
+      this.trigger("error", noRepErr);
```

Comment:
_Note_: I don't remember if we exchanged about this, but this means that we throw even if the application e.g. disable the video track on a `noPlayableTrack` event for video content - where it could theoretically play under the wanted settings.

I'm OK with this behavior, it's just to check whether we're on the same page on this.